src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { fetchArticleById } from '../api';
+
+const SingleArticle = () => {
+    const [article, setArticle] = useState({});
+    const { article_id } = useParams();
+
+    useEffect(() => {
+        fetchArticleById(article_id).then((articleData) => {
+          setArticle(articleData);
+        });
+      }, [article_id]);
+
+    return (
```

Comment:
Please give a visual feedback to the user of the articles loading. Use conditional rendering to achieve that.